### PR TITLE
Change static factory methods for DocumentPath

### PR DESCRIPTION
**Pull request by kuujo**
Construct a root path when an array/list of nodes is provided. Based on discussion with @aruanruan

### CODE REVIEW

File: core/src/main/java/io/atomix/core/tree/DocumentPath.java [hunk 1]
```diff
@@ -36,20 +35,67 @@
...
+
+  /**
+   * Creates a new root {@code DocumentPath} from a list of path elements.
+   *
+   * @param elements path elements
+   * @return {@code DocumentPath} instance
+   */
+  public static DocumentPath from(List<String> elements) {
+    List<String> copy = new ArrayList<>();
+    copy.add("");
```
Comment (johnou) [inline core/src/main/java/io/atomix/core/tree/DocumentPath.java]: what's the empty string for?

Comment (johnou) [reply core/src/main/java/io/atomix/core/tree/DocumentPath.java]:
```java
  /**
   * Root document path.
   */
  private static final String ROOT_PATH = "/";

  /**
   * Root document tree path.
   */
  public static final DocumentPath ROOT = new DocumentPath(ImmutableList.of(""));
```
do these still match?

Comment (johnou) [reply core/src/main/java/io/atomix/core/tree/DocumentPath.java]: just read over the converation on gitter, might be nicer if we introduce a constant here, ROOT_ELEMENT, self documenting code ftw.

Comment (kuujo) [reply core/src/main/java/io/atomix/core/tree/DocumentPath.java]: Good plan

**Pull request merged**